### ISSUE TITLE
Custom Commands UI to accept new user inputs.

### ISSUE THREAD

**Issue by advaitjavadekar** [ForkIssues/TrackerWin#1422]
Can the **Custom Commands** have an option so that after running one they can accept some new user inputs? Right now it looks like custom commands can only run fixed scripts. 
For eg. 
Creating a new Custom command for switching between branches and updating submodules. 
_Switch with Sub_ -> `switch --recurse-submodules`
User Inputs: this/is/new_branch on running _Switch with Sub_

Alternatively is there already a way to open bash/console from some point within the script using custom commands?

**Comment by db2222** [comment]
```Custom Commands``` are such a cool feature of Fork!
Having user input is the only missing piece to make it absolutely perfect 😀

It would be ideal if it were possible either through UI or Action (script).
Furthermore allowing multiple inputs would be very nice.
For the UI variant multi-line would be very handy (converted to ```\n``` in the string for the actual usage).

My guess is that it would be easier to implement for the UI variant 🙂
Maybe add an area under the ```Buttons``` in ```Custom Commands```:
```Input 1```: ```[Label for input]``` ```[+ button to add a further input field]```

The amount of inputs should probably be limited (max. 5?) and/or the ```Custom Commands``` dialogue should be scrollable.

Whilst executing the custom command these input fields then could be shown in the popup above the buttons:
```Label for input 1```: ```[Text area]```
...
```OK button``` ```Cancel button```
The popup window should also be scrollable.

The entered input then could be used via variables named $input1, $input2 etc. in the script.

Just as suggestions 🙂

Edit: Noticed that I got some things muddled up. Sorry 😅

**Comment by db2222** [comment]
@DanPristupov @TanyaPristupova 
Is there any chance that this could be implemented in the future? 🙂

This would open so many possibilities and immediately solve other (even future) tickets.
Just at the top of my head it would be a direct solution for:
- #1581
- #1217/#1346 if somebody wanted to create a new branch based on the result of ```merge-base``` (input new branch name).

Thanks so much in advance if this gets considered.
And thank you so much for your hard work on Fork. It's easily my favourite git UI tool!

**Comment by DanPristupov** [comment]
@db2222 we are working on that right now.

Most likely there will be multiple types of fields: branch/remotebranch/tag dropdown, file path selector, branch text field, generic text field

The fields will be accessible like `$1{SHA}` or `$2{filepath}`

**Comment by db2222** [comment]
@DanPristupov That sounds awesome! Thank you.

**Comment by db2222** [comment]
@DanPristupov Just out of curiosity. Is this being developed? 🙂

**Comment by DanPristupov** [comment]
Yes, it's almost done, but it's not released, because currently we are finishing internal refactoring (a technical debt) which took longer that we expected.

**Comment by DanPristupov** [comment]
Finally! 🎁

Recently (Fork 1.84) we created a new, more flexible syntax for custom commands variables. And today (Fork 1.85) we introduced UI custom commands.

If you have a chance, please try it and share some feedback!

**Comment by db2222** [comment]
@DanPristupov It looks very promising 😀 Thank you. I will try to check/test it closer the next couple of days.

**Comment by db2222** [comment]
@DanPristupov I had to do a little test immediately 😉 One thing I noticed is that it would be very nice if there also would be a text area field (so multiple lines) as a control type. I personally would need it to do #1581.

**Comment by DanPristupov** [comment]
Yes, I agree. This is just the initial release. It was already delayed for too long.

**Comment by Otiel** [comment]
This is great! 🎉 
Is it possible to specify multiple filters for the same control? 

For instance, I'd like to provide a branch selector control based on several filters : `refs/remote/origin/develop;refs/remote/origin/main`

**Comment by DanPristupov** [comment]
@Otiel Currently it's not possible but I think we should add that.

The only difference, we must use space ` ` instead of `;` as a separator
(same as in git for-each-ref pattern: `git for-each-ref refs/heads/develop refs/heads/main`
